examples/example_hyper.rs

```diff
@@ -13,72 +13,78 @@
 
 extern crate prometheus;
 extern crate hyper;
+#[macro_use]
+extern crate lazy_static;
 
-use std::thread;
-use std::time::Duration;
+use std::time::Instant;
 
 use hyper::header::ContentType;
 use hyper::server::{Server, Request, Response};
 use hyper::mime::Mime;
 
 use prometheus::encoder::{Encoder, TextEncoder};
-use prometheus::{Counter, Gauge, Opts, Registry};
+use prometheus::{Counter, Opts, Gauge, Histogram, HistogramOpts};
 
-fn main() {
-    let counter_opts =
-        Opts::new("test_counter", "test help").const_label("a", "1").const_label("b", "2");
-    let counter = Counter::with_opts(counter_opts).unwrap();
-
-    let gauge_opts = Opts::new("test_gauge", "test help").const_label("answer", "42");
-    let gauge = Gauge::with_opts(gauge_opts).unwrap();
-
-    let r = Registry::new();
-    r.register(Box::new(counter.clone())).unwrap();
-    r.register(Box::new(gauge.clone())).unwrap();
-
-    counter.inc();
-    assert_eq!(counter.get() as u64, 1);
-    counter.inc_by(42.0).unwrap();
-    assert_eq!(counter.get() as u64, 43);
-
-    let c2 = counter.clone();
-    thread::spawn(move || {
-        loop {
-            thread::sleep(Duration::from_millis(300));
-            c2.inc();
-        }
-    });
-
-    thread::spawn(move || {
-        loop {
-            thread::sleep(Duration::from_millis(500));
-            gauge.inc();
-
-            thread::sleep(Duration::from_millis(500));
-            gauge.dec();
-
-            thread::sleep(Duration::from_millis(500));
-            gauge.set(42.0);
-        }
-    });
+lazy_static! {
+    static ref HTTP_COUNTER: Counter = {
```

Comment:
can we define a metric to do this, like register_counter or other, so people only need to pass a corresponding options. Like

```
register_counter!(HTTP_COUNTER, opts);
```

Then we can use HTTP_COUNTER any where.